torch/nn/modules/transformer.py

```diff
@@ -203,12 +203,16 @@ def forward(self, src: Tensor, mask: Optional[Tensor] = None, src_key_padding_ma
         output = src
         convert_to_nested = False
         first_layer = self.layers[0]
+        src_key_padding_mask_for_layers = src_key_padding_mask
         if isinstance(first_layer, torch.nn.TransformerEncoderLayer):
             if (not first_layer.norm_first and not first_layer.training and
                     first_layer.self_attn.batch_first and
                     first_layer.self_attn._qkv_same_embed_dim and first_layer.activation_relu_or_gelu and
                     first_layer.norm1.eps == first_layer.norm2.eps and
-                    src.dim() == 3 and self.enable_nested_tensor) :
+                    src.dim() == 3 and self.enable_nested_tensor and
+                    src_key_padding_mask is not None and
+                    # is src_key_padding right polarity or needs inversion?
```

Comment:
You have it correct here - `src_key_padding` needs inversion.

The mask passed to `_nested_tensor_from_mask_left_aligned` is semantically defined as what to *include* within the NT, and `src_key_padding` specifies locations of padding, which should be *excluded*.